lib/IDE/IDERequests.cpp

```diff
@@ -995,16 +995,24 @@ bool RangeResolver::walkToDeclPre(Decl *D, CharSourceRange Range) {
 }
 
 bool RangeResolver::walkToExprPost(Expr *E) {
+  if (!Impl->shouldEnter(E))
+    return false;
   Impl->leave(E);
   return !Impl->hasResult();
 }
 
 bool RangeResolver::walkToStmtPost(Stmt *S) {
+  if (!Impl->shouldEnter(S))
+    return false;
   Impl->leave(S);
   return !Impl->hasResult();
 };
 
 bool RangeResolver::walkToDeclPost(Decl *D) {
+  if (D->isImplicit())
+    return false;
+  if (!Impl->shouldEnter(D))
+    return false;
```

Comment:
I think maybe this should be true? I think in the pose the result is used to terminate the walk or not.

Reply:
Good point - all the additions in the post should return true, thanks.